### DISCUSSION TITLE
Exception ao rodar um dos testes

### DISCUSSION THREAD

**Discussion by oGrandeK** [balta-io/forum#302] [Tira Dúvidas] [answered]
Boa noite professor, em um dos testes dentro do StudentTests.cs mais especificamente no ShouldReturnErrerWhenSubscriptionHasNoPayment() quando eu rodo os teste ele lança uma exception, o que devo fazer?

![image](https://user-images.githubusercontent.com/89154726/230236151-ca873ebb-4f16-4698-a5dd-d4956a149178.png)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @oGrandeK, como vai?

Vai no método AddSubscription e coloca um breakpoint... deve ter alguma validação lá dentro que está dando errada e tornando o IsValid false.

Outra coisa que você pode fazer é inspecionar o objeto _student e ver dentro do Notifications se tem alguma informação lá.

**Reply by oGrandeK** [reply]
Boa noite, meu _student está com 0 notificações.

![image](https://user-images.githubusercontent.com/89154726/231016701-ba05694a-babb-4c55-8bf6-06c188aa5aea.png)

e segue a mesma Exception

![image](https://user-images.githubusercontent.com/89154726/231016815-7d38c906-4a81-45e8-a6d9-f874fe0d7554.png)

**Reply by oGrandeK** [reply]
Boa noite novamente, fiz a alteração do `Assert.IsTrue(!_student.IsValid)` para `AreEqual(false, !_student.IsValid)` e o teste passou, continuo sem saber se a lógica está certa ou não rs kkkkkk

![image](https://user-images.githubusercontent.com/89154726/231019756-9d7ccd37-99f5-4daa-b0e2-b53da623d122.png)
